### DISCUSSION TITLE
Docx to Markdown: write tables as HTML

### DISCUSSION THREAD

**Discussion by denismaier** [jgm/pandoc#10288] [Q&A]
When converting docx to Markdown, us it possible to use Html for the tables? I'm having tables where the Markdown table syntax is hard to read, and I'd prefer to work with html tables.

**Comment by massifrg** [comment]
Maybe a lua filter like this could work (I did not test it):

```lua
return {
  {
    Table: function(t)
      if FORMAT == 'markdown' then
        local doc = pandoc.Pandoc( { t } )
        local html = pandoc.write(doc, 'html')
        return pandoc.RawBlock('markdown', html)
      end
    end,
  }
}
```

**Reply by denismaier** [reply]
Cool thanks. I'll also need to make sure the html tables get interpreted as tables when converting to other formats than HTML, but @tarleb seems to have already written a [filter](https://tex.stackexchange.com/a/631272) for that.

**Reply by denismaier** [reply]
And I still need to test your filter....

**Reply by denismaier** [reply]
Corrected filter:

```lua
return {
  {
    Table = function(t)
      if FORMAT == 'markdown' then
        local doc = pandoc.Pandoc( { t } )
        local html = pandoc.write(doc, 'html')
        return pandoc.RawBlock('markdown', html)
      end
    end,
  }
}
```

**Comment by HeirOfNorton** [comment]
The filter shouldn't be necessary. Tables in Markdown are extensions, so you just have to disable the table extensions in the pandoc command and it should automatically convert tables to HTML. Like so: `pandoc -t markdown-simple_tables-multiline_tables-grid_tables-pipe_tables -o output.md input.docx`. It's kind of a long bit to add to the command line since you have to disable all four types of markdown tables, but it works.

**Comment by denismaier** [comment]
The filter and the `-t markdown-simple_tables-multiline_tables-grid_tables-pipe_tables` produce essentially the same result.

However, I've noticed one thing: both solutions write the tables like so:

```markdown
Bla bla bla.

<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th>1</th>
<th>2</th>
<th>3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4</td>
<td>5</td>
<td>6</td>
</tr>
</tbody>
</table>

Bla bla bla
```

I.e., there are no rawBlock markers, and without those the table will not be picked up by this filter:

```lua
function RawBlock (raw)
  return raw.format:match 'html'
    and pandoc.read(raw.text, 'html').blocks
    or raw
end
```

Further processing only works with this:

```markdown
Bla bla bla.

~~~{=html}
<table>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th>1</th>
<th>2</th>
<th>3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4</td>
<td>5</td>
<td>6</td>
</tr>
</tbody>
</table>
~~~

Bla bla bla
```

Is there a way to get these rawBlock attributes automatically?

**Reply by massifrg** [reply]
Try this:

```lua
return {
  {
    Table = function(t)
      if FORMAT == 'markdown' then
        local doc = pandoc.Pandoc( { t } )
        local html = pandoc.write(doc, 'html')
        return pandoc.List({
          pandoc.RawBlock('markdown', '~~~{=html}'),
          pandoc.RawBlock('markdown', html),
          pandoc.RawBlock('markdown', '~~~'),
        })
      end
    end,
  }
}
```

or this:

```lua
return {
  {
    Table = function(t)
      if FORMAT == 'markdown' then
        local doc = pandoc.Pandoc( { t } )
        local html = pandoc.write(doc, 'html')
        return pandoc.RawBlock('markdown', '~~~{=html}\n' .. html .. '\n~~~'),
      end
    end,
  }
}
```

**Reply by denismaier** [reply]
Great. That works.
